### PR TITLE
Added currentpage amd movepage to jobstore, added a button for previo…

### PR DESCRIPTION
**Pull request by jteshale**
Added currentPage and movePage to JobStore, added a button for previous and next. Able to see the number of jobs in one page out of all the jobs. It works for the first page, but when I click next, it returned every job.

### CODE REVIEW

File: src/containers/Home.tsx [hunk 1]
```diff
@@ -82,6 +85,14 @@ const Home = () => {
                     </Box>
                 </HStack>
             </List>
+        </Box>
+        <Box>
+            <HStack>
+                <Text>Showing 1-{jobs?.length} Jobs of {allJobs?.length} results</Text>
+                <Spacer />
+                <Button onClick={() => movePage(currentPage)}>Previous</Button>
```
Comment (antxaddis) [inline src/containers/Home.tsx:93]: This has to be `movePage(currentPage-1)`. Disable if currentPage is equal to 1.

File: src/containers/Home.tsx [hunk 2]
```diff
@@ -82,6 +85,14 @@ const Home = () => {
...
+                <Button onClick={() => movePage(currentPage)} width={'90px'}>Next</Button>
```
Comment (antxaddis) [inline src/containers/Home.tsx:94]: Same as above but with `currentPage+1`

File: src/store/JobStore.tsx [hunk 3]
```diff
@@ -49,6 +49,8 @@ export type JobContextValue = {
   isFetchingCurrentJob: boolean;
   fetchCurrentJob: (id: string) => void;
   currentJob: IJob | undefined;
+  currentPage: any;
```
Comment (antxaddis) [inline src/store/JobStore.tsx:52]: This should be a `number` type.

File: src/store/JobStore.tsx [hunk 4]
```diff
@@ -63,6 +65,8 @@ const JobProvider = ({ children }: any) => {
   const [isFetchingAllJobs, setIsFetchingAllJobs] = useState(false);
   const [isFetchingCurrentJob, setIsFetchingCurrentJob] = useState(false);
   const [currentJob, setCurrentJob] = useState<IJob | undefined>(undefined);
+  const [currentPage, setCurrentPage] = useState<any>('1');
```
Comment (antxaddis) [inline src/store/JobStore.tsx:68]: should be number type, and set to `1` without quotes.

File: src/store/JobStore.tsx [hunk 5]
```diff
@@ -118,6 +122,16 @@ const JobProvider = ({ children }: any) => {
         setIsFetchingAllJobs(false); 
       });
   }
+
+  const movePage = (pageNumber: number) => {
+    setCurrentPage(true);
```
Comment (antxaddis) [inline src/store/JobStore.tsx:127]: set to the pageNumber

File: src/store/JobStore.tsx [hunk 6]
```diff
@@ -118,6 +122,16 @@ const JobProvider = ({ children }: any) => {
...
+    setJobs(allJobs)
+    const page_size = 100;
+    const end = pageNumber * page_size;
+    const start = end - page_size;
+    console.log('pageNumber', pageNumber);
+    return allJobs?.slice(start, end);
```
Comment (antxaddis) [inline src/store/JobStore.tsx:133]: Don't return but intsead set the value of this slice to `setJobs`

**Comment by jteshale** [comment]
I fix as much as I could with the comments that you gave me. However, there is a problem with the Math.ceil(allJobs?.length / 100). There is an error for allJobs?.length saying Object is possibly 'undefined'. This is for the disable for the button next. I need to fix this to check if the pagination is completely done.

**Pull request merged**